### PR TITLE
Prepare the product-deploy for ServiceLayer.

### PR DESCRIPTION
**Pull request by CZYK**
resolves studyportals/product-deploy/issues/48
resolves studyportals/product-deploy/issues/11 - exposes `deploy.servicelayer.codebuild` task
resolves studyportals/product-deploy/issues/44

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# @studyportals/product-deploy@v2.1.4
+# @studyportals/product-deploy@v2.1.5-0
```
Comment (CZYK) [inline README.md]: Bump at least minor

File: README.md [hunk 2]
```diff
@@ -26,11 +33,75 @@ Toolset to deploy StudyPortals products
...
+| --- | --- | --- |
+| opts | <code>Object</code> |  | 
+| opts.to | <code>string</code> |  | 
+| [opts.from] | <code>string</code> | <code>&quot;current working directory&quot;</code> | 
+| [opts.gulp] | <code>Gulp</code> |  | 
+
+<a name="Deploy+assemble"></a>
+
+### deploy.assemble() ⇒ <code>Promise</code>
+Assemble the portal into the deploy location.
```
Comment (CZYK) [inline README.md]: Complete description

File: README.md [hunk 3]
```diff
@@ -26,11 +33,75 @@ Toolset to deploy StudyPortals products
...
+
+**Kind**: instance method of [<code>Deploy</code>](#Deploy)  
+<a name="Deploy+configure"></a>
+
+### deploy.configure() ⇒ <code>Promise</code>
+Copy the configuration.
```
Comment (CZYK) [inline README.md]: Complete description

File: README.md [hunk 4]
```diff
@@ -26,11 +33,75 @@ Toolset to deploy StudyPortals products
...
+
+**Kind**: instance method of [<code>Deploy</code>](#Deploy)  
+<a name="Deploy+composer"></a>
+
+### deploy.composer() ⇒ <code>Promise</code>
+Install _composer dependencies
```
Comment (CZYK) [inline README.md]: Complete description

File: README.md [hunk 5]
```diff
@@ -26,11 +33,75 @@ Toolset to deploy StudyPortals products
...
+
+**Kind**: instance method of [<code>Deploy</code>](#Deploy)  
+<a name="Deploy+prepare"></a>
+
+### deploy.prepare() ⇒ <code>Promise</code>
+Make sure the deploy location exists and is empty.
```
Comment (CZYK) [inline README.md]: Complete description

File: README.md [hunk 6]
```diff
@@ -26,11 +33,75 @@ Toolset to deploy StudyPortals products
...
+
+**Kind**: instance method of [<code>Deploy</code>](#Deploy)  
+<a name="Deploy+workingCopy"></a>
+
+### deploy.workingCopy() ⇒ <code>Promise</code>
+Deploy based on the current folder.
```
Comment (CZYK) [inline README.md]: Complete description

File: lib/simple-page-tester.js [hunk 7]
```diff
@@ -31,14 +40,14 @@ module.exports = class SimplePageTester{
 			this.retries(2);
 			this.timeout(30000);
 
-			pages.forEach(path => {
+			pages.forEach(path =>{
 
-				it(`${path} should return a 200`, done => {
+				it(`${path} should return a 200`, done =>{
```
Comment (iliqportals) [inline lib/simple-page-tester.js]: [Idea/Issue] 1 Flaw in the flow is 503's. They should be retried

File: lib/simple-page-tester.js [hunk 8]
```diff
@@ -58,21 +67,13 @@ module.exports = class SimplePageTester{
...
-							logger(repo, failed, pages.length, () => done());
-							return;
-						}
-
-						done();
-					})
-				});
+				return logger(repo, failed, pages.length)
+					.catch(ex =>{
+						throw ex;
```
Comment (iliqportals) [inline lib/simple-page-tester.js]: [CodeStyle] empty new line after {

File: lib/simple-page-tester.js [hunk 9]
```diff
@@ -1,28 +1,37 @@
...
+	 * @param {String} url - The base url to test.
+	 */
 	constructor(repo, url){
 
 		this.repo = repo;
 		this.url = url;
 	}
 
+	/**
+	 * Run the test
```
Comment (iliqportals) [inline lib/simple-page-tester.js]: [Text] Executes simple accessbility tests to the specified pages.
If the pages cannot be reached, the test will be marked as failed.

**Pull request merged**